sql/core/src/main/scala/org/apache/spark/sql/execution/stat/StatFunctions.scala

```diff
@@ -33,6 +34,31 @@ private[sql] object StatFunctions extends Logging {
     counts.Ck / math.sqrt(counts.MkX * counts.MkY)
   }
 
+  /** Calculate the Pearson Correlation matrix for given DataFrame */
+  private[sql] def pearsonCorrelation(df: DataFrame): DataFrame = {
+    val fieldNames = df.schema.fieldNames
+    val dfStructAttrs = ArrayBuffer[AttributeReference](
+              AttributeReference("FieldName", StringType, true)())
+    val rows = fieldNames.map{fname => val countsRow = new GenericMutableRow(fieldNames.length + 1)
+        countsRow.update(0, UTF8String.fromString(fname))
+        countsRow
+    }.toSeq
+    // generates field types of the output DataFrame
+    for(field <- fieldNames) dfStructAttrs += AttributeReference(field, DoubleType, true)()
+
+    // fills the correlation matrix by computing column-by-column correlations
+    for (i <- 0 to fieldNames.length - 1){
+      for (j <- 0 to i){
+          val corr = pearsonCorrelation(df, Seq(fieldNames(i), fieldNames(j)))
```

Comment:
I'm not sure if showing null is valid. If not numeric then not showing anything, I think so ....